libraries/Microsoft.Bot.Builder.Skills/ChannelApiMiddleware.cs

```diff
@@ -0,0 +1,138 @@
+                await CallChannelApiAsync(turnContext, next, invokeArgs, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                // just pass through
+                await next(cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private async Task CallChannelApiAsync(ITurnContext turnContext, NextDelegate next, ChannelApiArgs invokeArgs, CancellationToken cancellationToken)
+        {
+            switch (invokeArgs.Method)
+            {
+                // Send activity(activity)
+                case ChannelApiMethods.SendToConversation:
+                    {
+                        var activityPayload = (Activity)invokeArgs.Args[0];
+                        if (activityPayload.Type == ActivityTypes.EndOfConversation)
+                        {
+                            await ProcessEndOfConversationAsync(turnContext, next, activityPayload, cancellationToken).ConfigureAwait(false);
+                            invokeArgs.Result = new ResourceResponse(id: Guid.NewGuid().ToString("N"));
+                            return;
+                        }
+
+                        invokeArgs.Result = await turnContext.SendActivityAsync(activityPayload, cancellationToken).ConfigureAwait(false);
+                        return;
+                    }
+
+                // Send activity(replyToId, activity)
+                case ChannelApiMethods.ReplyToActivity:
+                    {
+                        var activityPayload = (Activity)invokeArgs.Args[1];
+                        activityPayload.ReplyToId = (string)invokeArgs.Args[0];
+
+                        if (activityPayload.Type == ActivityTypes.EndOfConversation)
+                        {
+                            await ProcessEndOfConversationAsync(turnContext, next, activityPayload, cancellationToken).ConfigureAwait(false);
+                            invokeArgs.Result = new ResourceResponse(id: Guid.NewGuid().ToString("N"));
+                            return;
+                        }
+
+                        invokeArgs.Result = await turnContext.SendActivityAsync(activityPayload, cancellationToken).ConfigureAwait(false);
+                        return;
+                    }
+
+                // UpdateActivity(activity)
+                case ChannelApiMethods.UpdateActivity:
+                    invokeArgs.Result = await turnContext.UpdateActivityAsync((Activity)invokeArgs.Args[0], cancellationToken).ConfigureAwait(false);
+                    return;
+
+                // DeleteActivity(activityId)
+                case ChannelApiMethods.DeleteActivity:
+                    await turnContext.DeleteActivityAsync((string)invokeArgs.Args[0], cancellationToken).ConfigureAwait(false);
+                    break;
+
+                // SendConversationHistory(history)
+                case ChannelApiMethods.SendConversationHistory:
```

Comment:
>case ChannelApiMethods.SendConversationHistory: [](start = 16, length = 47)

do we really want to let a skill request the full history of its parent bots conversations? This seems like a PII leak in the making.

Reply:
I am considering commenting this and other messages that we don't know yet if we want to support in skills and just leave the ones we do really need. We can always go back and add these if we find a scenario that justifies having it. Makes sense?